### DISCUSSION TITLE
Unable to install packages through yarn for v9

### DISCUSSION THREAD

**Discussion by fagom** [pmndrs/react-spring#1418] [Support]
Hi Guys,

I tried to follow the steps in the **Contributing.md** Document. But on Step 3, while trying to install all the packages through Yarn, i seem to be getting the below error. Not sure if this is version specific issue or I am doing something wrong.

### **Error Message:**

yarn install v1.22.10
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] Resolving packages...
success Already up-to-date.
$ yarn build && node ./scripts/prepare.js
yarn run v1.22.10
$ bic
SyntaxError: Invalid regular expression: /^(?:D:\programming\github\react-spring\demo\node_modules|D:\programming\github\react-spring\demo\\.DS_Store|D:\programming\github\react-spring\demo\dist|D:\programming\github\react-spring\demo\dist-ssr|D:\programming\github\react-spring\demo\([^/]+)\.local)$/: Unmatched ')'
    at new RegExp (<anonymous>)
    at matchAny (D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:87:32)
    at Object.createMatcher (D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:129:39)
    at GitIgnore.test (D:\programming\github\react-spring\node_modules\build-if-changed\lib\gitignore.js:46:39)
    at Object.opts.filter (D:\programming\github\react-spring\node_modules\build-if-changed\lib\cli.js:19:46)
    at filter (D:\programming\github\react-spring\node_modules\build-if-changed\lib\index.js:15:25)
    at crawl (D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:68:46)
    at async crawl (D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:64:25)
    at async D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:79:9
    at async Object.exports.run (D:\programming\github\react-spring\node_modules\build-if-changed\lib\cli.js:21:26)
Done in 0.21s.
SyntaxError: Invalid regular expression: /^(?:(.+/)?packages\([^/]+)\package\.json|(.+/)?targets\([^/]+)\package\.json)$/: Unmatched ')'
    at new RegExp (<anonymous>)
    at matchAny (D:\programming\github\react-spring\node_modules\recrawl-sync\lib\recrawl.js:87:32)
    at createMatcher (D:\programming\github\react-spring\node_modules\recrawl-sync\lib\recrawl.js:128:39)
    at recrawl (D:\programming\github\react-spring\node_modules\recrawl-sync\lib\recrawl.js:15:18)
    at exports.crawl (D:\programming\github\react-spring\node_modules\recrawl-sync\lib\recrawl.js:12:38)
    at readPackages (D:\programming\github\react-spring\scripts\prepare.js:27:3)
    at prepare (D:\programming\github\react-spring\scripts\prepare.js:42:20)
Done in 1.58s.

**Comment by joshuaellis** [comment]
Not sure what to tell you, we don't have a package-lock.json in the project.

**Reply by fagom** [reply]
Hi @joshuaellis 
Apologies for the trivial error. Tried re-cloning the repo and follow the same steps as before, but I am still getting the same error.
Sharing the output below.

yarn install v1.22.10
[1/4] Resolving packages...
[2/4] Fetching packages...
warning Pattern ["@react-spring/types@link:packages/types"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-types-9.1.1\\node_modules\\@react-spring\\types" as pattern ["@react-spring/types@9.1.1","@react-spring/types@9.1.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["@react-spring/web@link:targets/web"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-web-9.1.1\\node_modules\\@react-spring\\web" as pattern ["@react-spring/web@9.1.1","@react-spring/web@9.1.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["@react-spring/shared@link:packages/shared"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-shared-9.1.1\\node_modules\\@react-spring\\shared" as pattern ["@react-spring/shared@9.1.1","@react-spring/shared@9.1.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["@react-spring/core@link:packages/core"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-core-9.1.1\\node_modules\\@react-spring\\core" as pattern ["@react-spring/core@9.1.1","@react-spring/core@9.1.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["@react-spring/konva@link:targets/konva"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-konva-9.1.1\\node_modules\\@react-spring\\konva" as pattern ["@react-spring/konva@9.1.1","@react-spring/konva@9.1.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["@react-spring/three@link:targets/three"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-three-9.1.1\\node_modules\\@react-spring\\three" as pattern ["@react-spring/three@9.1.1","@react-spring/three@9.1.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["@react-spring/zdog@link:targets/zdog"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-zdog-9.1.1\\node_modules\\@react-spring\\zdog" as pattern ["@react-spring/zdog@9.1.1","@react-spring/zdog@9.1.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["@react-spring/native@link:targets/native"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-native-9.1.1\\node_modules\\@react-spring\\native" as pattern ["@react-spring/native@9.1.1","@react-spring/native@9.1.1"]. This could result in non-deterministic behavior, skipping.
warning Pattern ["@react-spring/animated@link:packages/animated"] is trying to unpack in the same destination "C:\\Users\\fagom\\AppData\\Local\\Yarn\\Cache\\v6\\npm-@react-spring-animated-9.1.1\\node_modules\\@react-spring\\animated" as pattern ["@react-spring/animated@9.1.1","@react-spring/animated@9.1.1"]. This could result in non-deterministic behavior, skipping.
info fsevents@2.2.1: The platform "win32" is incompatible with this module.
info "fsevents@2.2.1" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@2.1.3: The platform "win32" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "win32" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "lerna > @lerna/version > @lerna/github-client > @octokit/rest > @octokit/plugin-request-log@1.0.2" has unmet peer dependency "@octokit/core@>=3".
warning " > @react-spring/konva@9.1.1" has unmet peer dependency "konva@>=2.6".
warning " > @react-spring/konva@9.1.1" has unmet peer dependency "react-konva@>=16.8".
warning " > @react-spring/native@9.1.1" has unmet peer dependency "react-native@>=0.58".
warning " > @react-spring/three-v5@9.1.1" has incorrect peer dependency "react@>=16.11".
warning " > @react-spring/three-v5@9.1.1" has unmet peer dependency "react-three-fiber@>=4.0".
warning " > @react-spring/three-v5@9.1.1" has unmet peer dependency "three@>=0.115".
warning " > @react-spring/three@9.1.1" has unmet peer dependency "@react-three/fiber@>=6.0".
warning " > @react-spring/three@9.1.1" has incorrect peer dependency "react@>=16.11".
warning " > @react-spring/three@9.1.1" has unmet peer dependency "three@>=0.126".
warning " > @react-spring/web@9.1.1" has unmet peer dependency "react-dom@>=16.8".
warning " > @react-spring/zdog@9.1.1" has unmet peer dependency "react-dom@>=16.8".
warning " > @react-spring/zdog@9.1.1" has unmet peer dependency "react-zdog@>=1.0".
warning " > @react-spring/zdog@9.1.1" has unmet peer dependency "zdog@>=1.0".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/core > react-dom@16.14.0" has incorrect peer dependency "react@^16.14.0".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/konva > react-konva@16.8.6" has incorrect peer dependency "react@16.8.x".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/konva > react-konva@16.8.6" has unmet peer dependency "react-dom@16.8.x".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/konva > react-konva@16.8.6" has incorrect peer dependency "konva@^3.2.3".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/native > react-native@0.58.6" has incorrect peer dependency "react@16.6.3".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three-v5 > react-three-fiber@4.2.21" has incorrect peer dependency "react@>=16.13".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three-v5 > react-three-fiber@4.2.21" has unmet peer dependency "react-dom@>=16.13".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three > @react-three/fiber@5.3.15" has incorrect peer dependency "react@>=17.0".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three > @react-three/fiber@5.3.15" has unmet peer dependency "react-dom@>=17.0".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/zdog > react-zdog@1.0.11" has incorrect peer dependency "zdog@>=1.1".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three-v5 > react-three-fiber > react-reconciler@0.25.1" has incorrect peer dependency "react@^16.13.1".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three-v5 > react-three-fiber > react-use-measure@2.0.3" has incorrect peer dependency "react@>=16.13".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three-v5 > react-three-fiber > react-use-measure@2.0.3" has unmet peer dependency "react-dom@>=16.13".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three > @react-three/fiber > react-reconciler@0.26.1" has incorrect peer dependency "react@^17.0.1".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three > @react-three/fiber > react-use-measure@2.0.4" has incorrect peer dependency "react@>=16.13".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three > @react-three/fiber > react-use-measure@2.0.4" has unmet peer dependency "react-dom@>=16.13".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > @react-spring/three > @react-three/fiber > use-asset@1.0.2" has incorrect peer dependency "react@>=17.0".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/web@9.1.1" has unmet peer dependency "react-dom@>=16.8".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/konva@9.1.1" has unmet peer dependency "konva@>=2.6".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/konva@9.1.1" has unmet peer dependency "react-konva@>=16.8".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/three@9.1.1" has unmet peer dependency "@react-three/fiber@>=6.0".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/three@9.1.1" has incorrect peer dependency "react@>=16.11".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/three@9.1.1" has unmet peer dependency "three@>=0.126".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/zdog@9.1.1" has unmet peer dependency "react-dom@>=16.8".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/zdog@9.1.1" has unmet peer dependency "react-zdog@>=1.0".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/zdog@9.1.1" has unmet peer dependency "zdog@>=1.0".
warning "workspace-aggregator-be19a98a-a7db-47c4-afdd-a452e01569ec > react-spring > @react-spring/native@9.1.1" has unmet peer dependency "react-native@>=0.58".
[4/4] Building fresh packages...
$ yarn build && node ./scripts/prepare.js
yarn run v1.22.10
$ bic
SyntaxError: Invalid regular expression: /^(?:D:\programming\github\react-spring\demo\node_modules|D:\programming\github\react-spring\demo\\.DS_Store|D:\programming\github\react-spring\demo\dist|D:\programming\github\react-spring\demo\dist-ssr|D:\programming\github\react-spring\demo\([^/]+)\.local)$/: Unmatched ')'
    at new RegExp (<anonymous>)
    at matchAny (D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:87:32)
    at Object.createMatcher (D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:129:39)
    at GitIgnore.test (D:\programming\github\react-spring\node_modules\build-if-changed\lib\gitignore.js:46:39)
    at Object.opts.filter (D:\programming\github\react-spring\node_modules\build-if-changed\lib\cli.js:19:46)
    at filter (D:\programming\github\react-spring\node_modules\build-if-changed\lib\index.js:15:25)
    at crawl (D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:68:46)
    at async crawl (D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:64:25)
    at async D:\programming\github\react-spring\node_modules\build-if-changed\node_modules\recrawl\lib\recrawl.js:79:9
    at async Object.exports.run (D:\programming\github\react-spring\node_modules\build-if-changed\lib\cli.js:21:26)
Done in 0.70s.
SyntaxError: Invalid regular expression: /^(?:(.+/)?packages\([^/]+)\package\.json|(.+/)?targets\([^/]+)\package\.json)$/: Unmatched ')'
    at new RegExp (<anonymous>)
    at matchAny (D:\programming\github\react-spring\node_modules\recrawl-sync\lib\recrawl.js:87:32)
    at createMatcher (D:\programming\github\react-spring\node_modules\recrawl-sync\lib\recrawl.js:128:39)
    at recrawl (D:\programming\github\react-spring\node_modules\recrawl-sync\lib\recrawl.js:15:18)
    at exports.crawl (D:\programming\github\react-spring\node_modules\recrawl-sync\lib\recrawl.js:12:38)
    at readPackages (D:\programming\github\react-spring\scripts\prepare.js:27:3)
    at prepare (D:\programming\github\react-spring\scripts\prepare.js:42:20)
Done in 194.59s.

**Reply by joshuaellis** [reply]
I think the issue is with one is in `recrawl-sync` and shown here – https://github.com/aleclarson/recrawl/issues/6. It basically means you can't build and prepare the packages, so working on the library itself will be problematic.

As unfortunate as it is, I only have a mac, so I can't really fix this issue. However you could either fix the issue or create a PR to replace `recrawl-sync` with an alternative that does work with Windows. I'd be cool with that change.

**Reply by fagom** [reply]
Got it. Thanks.
